### PR TITLE
[FIX] product_weight: template variant

### PR DESCRIPTION
**Pull request by chienandalu**
If the variants for a template are archived a key error raises. Catch
the exception with a more informative message and try to determine the
proper variant in advance anyway.

cc @Tecnativa TT23823

### CODE REVIEW

File: product_weight/wizard/product_weight_update.py [hunk 1]
```diff
@@ -54,7 +55,11 @@ def default_get(self, fields):
     def calculate_product_bom_weight(self, bom, product=False):
         product_tmpl = bom.product_tmpl_id
         if not product:
-            product = product_tmpl.product_variant_ids[0]
+            product = product_tmpl.product_variant_ids[:1]
+            if not product:
+                raise UserError(_(
+                    'Missing variant for product %s' % (
```
Comment (pedrobaeza) [inline product_weight/wizard/product_weight_update.py]:
```suggestion
                    'Missing active variant for product %s' % (
```

**Comment by chienandalu** [comment]
Changes done

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Let's merge this old one:

/ocabot merge patch

**Comment by pedrobaeza** [comment]
Cherry-picked for v13 in 66083605aab8a907754d0b9115e4a23027e8aa8a

**Pull request merged**